src/Core/WinSWCore/LogAppenders.cs

```diff
@@ -470,21 +475,78 @@ private void ZipFiles(string directory, string fileExtension, string zipFileBase
                     string sourceFileName = Path.GetFileName(path);
                     string zipFilePattern = fileInfo.LastAccessTimeUtc.ToString(ZipDateFormat);
                     string zipFilePath = Path.Combine(directory, $"{zipFileBaseName}.{zipFilePattern}.zip");
-                    ZipOneFile(path, sourceFileName, zipFilePath);
-
-                    File.Delete(path);
+                    bool zipResult = ZipOneFile(path, sourceFileName, zipFilePath);
+                    if (zipResult)
+                    {
+                        File.Delete(path);
+                    }
                 }
             }
             catch (Exception e)
             {
                 EventLogger.LogEvent($"Failed to Zip files. Error {e.Message}");
             }
+
+            if(!(ZipDaysToKeep is null) && ZipDaysToKeep != 0)
+            {
+                try
+                {
+                    //Get all the zipfiles that might match the naming pattern
+                    List<string> ZipFiles = new List<string>(Directory.GetFiles(directory, zipFileBaseName + ".*.zip"));
+                    //Now delete the ones we don't want to keep, skipping any that don't match the naming pattern
+                    foreach ( string zipFilePath in ZipFiles)
+                    {
+                        if (File.Exists(zipFilePath))
+                        {
+                            //ZipDateFormat could contain literal characters, including the '.', so this gets a bit tricky

```

Comment:
Doesn't `TryParseExact` handle `.`? It could be a real problem if `zipFileBaseName` contains `.`.

Reply:
TryParseExact should handle '.', but doesn't seem to like the full file name. Implemented a fix for the possibility of  ZipFileBaseName containing '.'